src/sentry/tasks/relay.py

```diff
@@ -65,13 +64,13 @@ def update_config_cache(
             keys = [ProjectKey.objects.get(public_key=public_key)]
         except ProjectKey.DoesNotExist:
             # In this particular case, where a project key got deleted and
-            # triggered an update, we at least know the public key that needs
-            # to be deleted from cache.
+            # triggered an update, we know that key doesn't exist and we want to
+            # avoid creating more tasks for it.
             #
             # In other similar cases, like an org being deleted, we potentially
             # cannot find any keys anymore, so we don't know which cache keys
             # to delete.
-            projectconfig_cache.delete_many([public_key])
+            projectconfig_cache.set_many({public_key: {"disabled": True}})
```

Comment:
If `{"disabled":True}` is now a valid cache value, do we also return it to Relay and can Relay handle it? Or should we suppress it on the endpoint?

Reply:
@jjbayer yes, this is the intended explicit way of informing Relay that a project is disabled. It was taken from the endpoint code here:
https://github.com/getsentry/sentry/blob/53232338eafe0ed0b9713e629fd8c56766db2f81/src/sentry/api/endpoints/relay/project_configs.py#L177

cc @untitaker, could you please double-check that writing this to the cache is sound?

Reply:
Relay should be able to handle it, we already use it for inactive projects: https://github.com/getsentry/sentry/blob/b76adec38aae204d54487fcac9921251d64ab164/src/sentry/tasks/relay.py#L84

Reply:
that's fine